python/lsst/multiprofit/fitting/fit_source.py

```diff
@@ -0,0 +1,897 @@
+                    if not config_comp.size_x.fixed:
+                        parameters[f"{prefix_comp}{label_size}_x"] = component.ellipse.size_x_param
+                    if not config_comp.size_y.fixed:
+                        parameters[f"{prefix_comp}{label_size}_y"] = component.ellipse.size_y_param
+                    if not config_comp.rho.fixed:
+                        parameters[f"{prefix_comp}rho"] = component.ellipse.rho_param
+                    if not config_comp.flux.fixed:
+                        # TODO: return this to component.integralmodel
+                        # when binding for g2f.FractionalIntegralModel is fixed
+                        params_flux = get_params_uniq(component, fixed=False, nonlinear=False)
+                        if len(params_flux) != n_channels:
+                            raise ValueError(f"{params_flux=} len={len(params_flux)} != {n_channels=}")
+                        for channel, param_flux in zip(self.channels, params_flux):
+                            parameters[f"{prefix_comp}{channel.name}_flux"] = param_flux
+                    if hasattr(config_comp, "sersic_index") and not config_comp.sersic_index.fixed:
+                        parameters[f"{prefix_comp}sersicindex"] = component.sersicindex_param
+
+        return parameters
+
+    @cached_property
+    def sources_priors(self) -> tuple[tuple[g2f.Source], tuple[g2f.Prior]]:
+        sources, priors = self.config.make_sources(channels=self.channels)
+        return tuple(sources), tuple(priors)
+
+
+class CatalogSourceFitterABC(ABC, pydantic.BaseModel):
+    """Fit a Gaussian mixture source model to an image with a PSF model.
+
+    Notes
+    -----
+    Any exceptions raised and not in errors_expected will be logged in a
+    generic unknown_flag failure column.
+    """
+
+    model_config = frozen_arbitrary_allowed_config
+
+    errors_expected: dict[Type[Exception], str] = pydantic.Field(
+        default_factory=dict,
+        title="A dictionary of Exceptions with the name of the flag column key to fill if raised.",
+    )
+    modeller: Modeller = pydantic.Field(
+        default_factory=Modeller,
+        title="A Modeller instance to use for fitting.",
+    )
+
+    @staticmethod
+    def _get_logger():
+        logging.basicConfig()
+        logger = logging.getLogger(__name__)
+        logger.level = logging.INFO
+
+        return logger
+
+    def copy_centroid_errors(
+        self,
+        columns_cenx_err_copy: tuple[str],
+        columns_ceny_err_copy: tuple[str],
+        results: Table,
+        catalog_multi: Sequence,
+        catexps: list[CatalogExposureSourcesABC],
+        config_data: CatalogSourceFitterConfigData,
+    ):
+        if columns_cenx_err_copy or columns_ceny_err_copy:
+            raise RuntimeError(
+                f"Fitter of {type(self)=} got {columns_cenx_err_copy=} and/or {columns_ceny_err_copy=}"
+                f" but has not overriden copy_centroid_errors"
+            )
```

Comment:
This is needed to support a specific use case:
- centroids are not fit but derived from the inputs (usually by copying an existing column in the multiband catalog)
- config.convert_cen_xy_to_radec is true
If both of those are true, then further down both RA/dec and errors thereof will be computed (in a rather stupid way, but it is what it is) every iteration, so these columns need to be added before fitting starts. There wasn't any other override-able method to support this, so I made a new and possibly too specifically-named one.

In meas_extensions_multiprofit, all this does is copy `slot_centroid_x/y` into results. I think I included catexps in case the single-band catalog were needed for some reason, or other exposure properties, and config in case the columns to copy from needed to be configured. It does actually allow the implementer to do as they see fit with any of the arguments, so maybe a less specific name is in order.